CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+<!--
+SPDX-FileCopyrightText: 2019 The curiu authors
+
+SPDX-License-Identifier: LicenseRef-ISC-curiu
+-->
+
+# Contributing
+
+- Add yourself to the [AUTHORS.md](AUTHORS.md) file.
+- Autoformat new/changed code with [`black`](https://github.com/psf/black).
+- Add SPDX copyright/license headers or `.license` files and check them via [`reuse lint`](https://github.com/fsfe/reuse-tool/tree/v0.5.0#usage).
+- Add informative commit messages:
+  - Use imperative language.
+  - Use short titles.
+  - If there are important or non-obvious details, mention them (succinctly) in the message
+    body.
+  - If possible, add references in the message body.
+    Use tags or commit checksums in those references rather than just pointing to a file on some
+    branch.
+  - If one commit does more than one thing, be sure to list the individual changes in the
+    message body.
+
+  e.g.:
+  ```
+  Add functionality X
+
+  Important details that are not covered in title or individual changes listed below.
+
+  - fix feature A
+    ref:
+      - https://github.com/conda/conda/blob/4.7.12/conda/resolve.py#L945-L951
+  - remove feature B
+  - add feature C
+    ref:
+      - Proposal of feature C:
+        https://github.com/curiu/curiu/issue/XXX
+
+  ref:
+  - Discussion of functionality X
+    https://github.com/curiu/curiu/issues/YYY
+    https://github.com/conda/conda/issues/7808
+  ```
+- Do not push directly to upstream branches:
+  - Fork the project.
+  - Push changes to a *new* branch on your fork.
+  - Submit a pull request to the upstream branch.
+
+- The project makes use of the [ISC License](LICENSES/LicenseRef-ISC-curiu.txt).
+  - Your contributions are expected to be submitted to be used under this license.
+  - If you want to submit a contribution under a different license, be sure to
+    - adjust the SPDX license/copyright information of those files accordingly,
+    - clearly point out that license in your pull request.
+
+- Code in this repository is meant to be submitted to [Conda](https://github.com/conda).
+  - **Please be sure to have signed the [Conda Contributor License Agreement](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement).**
+  - **If you do not want or are not able to sign the Conda CLA, you can only submit contributions
+    to the `external` directory. See [external/README.md](external/README.md) for details.**
+  - **!!!**<br />
+    **TODO: Check how we can contribute to Conda under the CLA.**
```

Comment:
Another (stupid) possibility is of course to always state that we submit this as "Not a Contribution." and handle it on a per case basis -- I guess no one's gonna be happy with that either xD.
https://github.com/conda/conda/blame/4.7.12/CONTRIBUTING.md#L149-L151
> excluding communication that is conspicuously marked or otherwise designated in writing by You as 
"Not a Contribution."

Reply:
I see a few ways to think about this:

1. Is curiu a (friendly) fork of conda?  If so, this can be seen as a long-running experimental branch.  Anyone contributing code to that branch needs to sign the CLA before that branch can be merged into Conda.
2. The external directory is an interesting way to handle conflicted contributions.  I'm not sure it will work well, though.  Anything in external then has to be really optional and not default, which stinks for good additions.
3. If curiu has a CLA that assigns rights or whatever from the contributor to the curiu organization, then I think curiu (or agents thereof) can submit PRs to conda, as long as they can act on behalf of curiu in assigning rights from curiu to conda.

IANAL.  I hate this stuff.  Having Curiu under the CC0 license seems like it should obviate all of this, though - it is already a disclaimer of copyright, right?  Anything submitted to curiu will become CC0, and thus there shouldn't be any further concern with conda's CLA.

Reply:
> IANAL. I hate this stuff.

Thank you for having taken look despite that :heart:.

1. friendly yes, not a fork (though I/we might fork `conda` to `github.com/curiu/conda` too at some point).
2. "external directory  \[...\] not sure it will work well \[...\] not default, which stinks for good additions." Of course, I agree. It's just an alternative to a never-(fully)-merged PR with potentially better visibility, to be honest. Hopefully always empty or removed in the future ;).
3. Yes, I believe given my current understanding of the Conda CLA, the easiest thing would be to have a short curiu CLA stating something like "by contributing to curiu, you allow anyone to submit your contributions to the Conda project under the Conda CLA *on your behalf*". Then we should be all set. (Well, at least for their "original" work/contributions, \*sigh\*...)

> Having Curiu under the CC0 license seems like it should obviate all of this, though

The "funny" (=extremely annoying) thing is, that from my understanding, it does not, sadly.
**NOTE: if you want to spare your sanity, stop reading and just forward this to someone at Anaconda that handles CLA/legal issues ;)**

The Conda CLA forces me to either submit *my* original work or be *allowed* to act on someone else's *behalf*. I assume that under a strict legal interpretation, the "on behalf" part might require "authorization".
Meaning if someone puts out work under CC0, they allow me to do (more or less) whatever I like with that work. But they do not allow me to do whatever I like with their *name*, i.e., I can't pretend to "act on their behalf" without permission. Ergo, I can't apply point 7 of the Conda CLA unless they give me permission to write "Submitted on behalf of a third-party: [named here]".
The first search result of
https://duckduckgo.com/html/?q=legal%20definition%20%22on%20behalf%20of%22
is
https://legal-dictionary.thefreedictionary.com/on+behalf+of
which lists some interpretations, including "be the authorized agent for, be the authorized represennative for" which confirm my hunch of of possible problematic/strict interpretation :/ .